lib/hubspot/config.rb

```diff
@@ -42,7 +42,11 @@ def reset!
         @portal_id = nil
         @logger = DEFAULT_LOGGER
         @access_token = nil
-        Hubspot::Connection.headers({})
+        @client_id = nil
+        @client_secret = nil
+        @redirect_uri = nil
+        
+        Hubspot::Connection.headers.clear
```

Comment:
Could alternatively be more selective via:

`Hubspot::Connection.headers.delete("Authorization")`